### PR TITLE
MINOR: add concurrent test for consumer.poll

### PR DESCRIPTION
**Pull request by TaiJuWu**
Add concurrent test for both consumer.poll

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java [hunk 1]
```diff
@@ -3382,7 +3382,43 @@ public void testCommittedThrowsTimeoutExceptionForNoResponse(GroupProtocol group
         assertEquals("Timeout of 1000ms expired before the last committed offset for partitions [test-0] could be determined. " +
             "Try tuning default.api.timeout.ms larger to relax the threshold.", timeoutException.getMessage());
     }
-    
+
+    @ParameterizedTest
+    @EnumSource(value = GroupProtocol.class)
+    public void testPreventMultiThread(GroupProtocol groupProtocol) throws InterruptedException {
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java:3388]: Could you please consider writing it by UT. https://github.com/apache/kafka/blob/4a3ab89f95aba294bb536af55548522d946d1ee3/clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java#L2428 is a good example

**Review by chia7712 — COMMENTED** [review]
@TaiJuWu thanks for increasing the test coverage

File: clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java (hunk 1 above)
Comment (TaiJuWu) [reply clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java:3388]: Hi @chia7712 , thanks for your suggestion! Updated it!

File: clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java [hunk 2]
```diff
@@ -3382,7 +3382,35 @@ public void testCommittedThrowsTimeoutExceptionForNoResponse(GroupProtocol group
...
+        Properties props = new Properties();
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java]: please remove unused variable

**Review by chia7712 — COMMENTED** [review]
@TaiJuWu thanks for your patch

File: clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java (hunk 2 above)
Comment (TaiJuWu) [reply clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java]: Sorry for this, removed.

**Review by chia7712 — APPROVED** [review]

**Pull request merged**